jax/_src/numpy/polynomial.py

```diff
@@ -102,15 +114,156 @@ def roots(p: ArrayLike, *, strip_zeros: bool = True) -> Array:
+
+  Jax implementation of :func:`numpy.polyfit`
+
+  Given a set of data points ``(x, y)`` and degree of polynomial ``deg``, the
+  function finds a polynomial equation of the form:
+
+  .. math::
+
+	   y = p(x) = p[0] x^{deg} + p[1] x^{deg - 1} + ... + p[deg]
+
+  Unlike :func:`numpy.polyfit` implementation of polyfit, :func:`jax.numpy.polyfit`
+  will not warn on rank reduction, which indicates an ill conditioned matrix.
+  Also, it works best on rcond <= 10e-3 values.
+
+  Args:
+    x: Array of data points of shape ``(M,)``.
+    y: Array of data points of shape ``(M,)`` or ``(M, K)``.
+    deg: Degree of the polynomials.
+    rcond: Relative condition number of the fit. Default value is ``len(x) * eps``.
+    full: Switch that controls the return value. Default is ``False`` which
+      restricts the return value to the array of polynomail coefficients ``p``.
+      If ``True``, the function returns a tuple ``(p, resids, rank, s, rcond)``.
+    w: Array of weights of shape ``(M,)``. If None, all data points are considered
+      to have equal weight. If not None, the weight :math:`w_i` is applied to the
+      unsquared residual of :math:`y_i - \widehat{y}_i` at :math:`x_i`, where
+      :math:`\widehat{y}_i` is the fitted value of :math:`y_i`. Default is None.
+    cov: Boolean or string. If ``True``, returns the covariance matrix scaled
+      by ``resids/(M-deg-1)`` along with ploynomial coefficients. If 
+      ``cov='unscaled'``, returns the unscaaled version of covariance matrix.
+      Default is ``False``.
+
+  Returns:
+    - An array polynomial coefficients ``p`` if ``full=False`` and ``cov=False``.
+
+    - A tuple of arrays ``(p, resids, rank, s, rcond)`` if ``full=True``. Where
+
+      - ``p`` is an array of shape ``(M,)`` or ``(M, K)`` containing the polynomial
+        coefficients.
+      - ``resids`` is the sum of squared residual of shape () or (K,).
+      - ``rank`` is the rank of the matrix ``x``.
+      - ``s`` is the singular values of the matrix ``x``.
+      - ``rcond`` as the array.
+    - A tuple of arrays ``(p, C)`` if ``full=False`` and ``cov=True``. Where
+
+      - ``p`` is an array of shape ``(M,)`` or ``(M, K)`` containing the polynomial
+        coefficients.
+      - ``C`` is the covariance matrix of polynomial coefficients of shape
+        ``(deg + 1, deg + 1)`` or ``(deg + 1, deg + 1, 1)``.
+
+  See Also:
+    - :func:`jax.numpy.poly`: Finds the polynomial coefficients of the given
+      sequence of roots.
+    - :func:`jax.numpy.polyval`: Evaluate a polynomial at specific values.
+    - :func:`jax.numpy.roots`: Computes the roots of a polynomial for given
+      coefficients.
+
+  Example:
+    >>> x = jnp.array([3., 6., 9., 4.])
+    >>> y = jnp.array([[0, 1, 2],
+    ...                [2, 5, 7],
+    ...                [8, 4, 9],
+    ...                [1, 6, 3]])
+    >>> p = jnp.polyfit(x, y, 2)
+    >>> with jnp.printoptions(precision=2, suppress=True):
+    ...     print(p)
```

Comment:
Use two-space indentations, here and below.

Reply:
Done